core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1909,41 +2267,41 @@ private void attachPartitionCheckFilesMatchMetadata(FilesFacade ff, Path path, R
                     case ColumnType.GEOSHORT:
                     case ColumnType.GEOINT:
                     case ColumnType.GEOLONG:
-                        attachPartitionCheckFilesMatchFixedColumn(ff, path, columnType, partitionSize, columnName, columnNameTxn);
+                        attachPartitionCheckFilesMatchFixedColumn(columnType, partitionSize, columnName, columnNameTxn);
                         break;
                     case ColumnType.STRING:
                     case ColumnType.BINARY:
-                        attachPartitionCheckFilesMatchVarLenColumn(ff, path, partitionSize, columnName, columnNameTxn);
+                        attachPartitionCheckFilesMatchVarLenColumn(partitionSize, columnName, columnNameTxn);
```

Comment:
This does not respect that the file can have a column top, it should adjust partitonSize by the columnTop